cluster/gce/upgrade.sh

```diff
@@ -464,13 +464,8 @@ function update-coredns-config() {
   cleanup() {
     rm -rf "${download_dir}"
   }
-  trap cleanup EXIT
+  trap cleanup RETURN
 
-  # Get the new installed CoreDNS version
-  echo "Waiting for CoreDNS to update"
-  until [[ $("${KUBE_ROOT}"/cluster/kubectl.sh -n kube-system get deployment coredns -o=jsonpath='{$.metadata.resourceVersion}') -ne ${COREDNS_DEPLOY_RESOURCE_VERSION} ]]; do
-     sleep 1
-  done
```

Comment:
out of curiosity, how did you determine this was the issue? was there a log in the existing test job runs that pointed here? how did you run the upgrade test to verify dropping this let the upgrades start working again?

Reply:
> how did you determine this was the issue? was there a log in the existing test job runs that pointed here?

when i ran this test locally, the test stuck here forever and the fact is the `resourceVersion` doesn't change, i.e. the version `COREDNS_DEPLOY_RESOURCE_VERSION ` before upgrade == the resource version after upgrade. removing this let this upgrade test working again. i didn't verify other upgrades.

> how did you run the upgrade test to verify dropping this let the upgrades start working again?

i modified https://github.com/kubernetes/kubernetes/blob/73b4f09dce7dfa0f66910483f7ce06a87dc31108/test/e2e/cloud/gcp/cluster_upgrade.go#L51-L51 with default to ` "v1.20.0-alpha.0.1462+b14fcc544bd5a1-bazel` to get around the hash issue and then ran the command `kubetest --build --up --test --test_args="--ginkgo.focus=\[Feature:BoundServiceAccountTokenVolume\]" --check-version-skew=false`

Reply:
> i modified kubernetes/test/e2e/cloud/gcp/cluster_upgrade.go with default to "v1.20.0-alpha.0.1462+b14fcc544bd5a1-bazel to get around the hash issue

we need to resolve that in a way that permits the automated job runs to succeed... a fix for the hash issue needs to be part of the kubernetes/kubernetes PR with the upgrade test fix described in https://github.com/kubernetes/kubernetes/pull/94835#pullrequestreview-489995883

Reply:
yes, i planned to submit a PR to fix the hash issue.